### DISCUSSION TITLE
Welcome to arduino-eclipse-plugin Discussions!

### DISCUSSION THREAD

**Discussion by jantje** [Sloeber/arduino-eclipse-plugin#1288] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we

**Comment by Catephatic** [comment]
Hi, I am running Sloeber Product on W10  and when I go Project->Properties->Arduino I am unable to get any USB ports listed for selection (only COM1) whatever Board or Upload Protocol I try. What do I need to do to get the USB ports offered?
Thanks

**Reply by jantje** [reply]
If there is only the port "COM1" it means Sloeber only finds "COM1". Sloeber always lists all found ports.
So the selection of the protocol does not change the list of com ports.
The ports are "searched" when the Project->Properties->Arduino view is opened. SO making changes to the connected boards will only be reflected when you close and reopen the Project->Properties

Note that
You can type anything as a com port but if it is not found by Sloeber it is very unlikely it will work out for you

**Comment by Catephatic** [comment]
Thank you very much for you quick reply. I have gone back to my older PC for the moment for my development, but I would very much like to move to the new PC if I can get uploading working. I am not sure how Sloeber makes the API call to poll the hardware peripherals, but for some reason on this PC (Dell Optiplex 7040 W10) the API call is returning nothing. I will try to do some more research to see if I can find an explanation.

**Reply by jantje** [reply]
The list of com ports on windows is fetched here
https://github.com/Sloeber/arduino-eclipse-plugin/blob/master/io.sloeber.core/src/io/sloeber/core/api/Serial.java#L122
As you can see it is a jssc call
The jssc library comes with sloeber https://github.com/Sloeber/arduino-eclipse-plugin/tree/master/io.sloeber.core/lib and is currently version 2.9.2 but the nightly is on 2.9.4 (so may contain fix)
If I would be investigating this issue I would start there.
What is the actual used jssc on my system?
Are there known/reported incompatibilities between jssc and the hardware I run
What are the jssc version 2.9.2 and later bugs
How do jssc versions an java versions relate
...

**Comment by ianr34** [comment]
the most likely thing is that you haven't installed the right usb driver - ie 340G or cp210X etc. Thus it isn't coming up as a serial port. 

ie first thing to check is in device manager to see if you actually have a com port for it to find... I've never know sloeber not to find one if it is there.

**Comment by darek-i-rafal-i-zbychu** [comment]
Sometimes it happens that despite connecting the programmer to the computer, it is not visible as a COM port. What is the reason?

The reason is usually that the system does not have the correct driver installed to support the programmer. In order for it to be available and visible as a virtual COM, you must install a VCP type driver.

**Comment by darek-i-rafal-i-zbychu** [comment]
I have been using Eclipse and Sloeber for a long time. I think they worked well until 2021. Since 2022 most versions have problems with the Eclipse Arduino plugin. I have tested a dozen or more different newer versions in the last time. Almost none of them work properly on Windows except the older ones. All of these versions display the error message: during compilation. Below is a screenshot:
![Eclipse_Error1](https://github.com/user-attachments/assets/55514ba2-55cd-4b91-a651-813742c467b9)

Has no one noticed this in the meantime? It is not the fault of the system or the computer, because I checked on a new, clean computer and different Windows systems (versions 7, 10, 11) and the error is the same. The error message does not allow you to guess what the problem is. 
Can anyone help solve this, point out what mistake I am making or describe the procedure for correctly installing Eclipse with the Arduino plugin?

**Reply by jantje** [reply]
Open project properties->sloeber->apply and close typically fixes path issues

**Comment by darek-i-rafal-i-zbychu** [comment]
I noticed this need to open the project properties a long time ago. However, it has no effect on the error that appears. This error occurs in all new versions, as I described earlier. Only Sloeber ver. 4.1 works on this computer, which contains the Arduino plugin installed by the manufacturer. Does this error not occur with you? I can't bypass it in any way. At first I thought it was the computer or system's fault, but it also occurs on other computers.

**Reply by jantje** [reply]
Using private hardware may cause this issue as well.

**Reply by jantje** [reply]
Private hardware like in sloeber in the menu  and select Preferences
![image](https://github.com/user-attachments/assets/815338c2-eae6-430d-9ef1-706a0648b071)

**Comment by darek-i-rafal-i-zbychu** [comment]
What does a private computer mean for Eclipse or Sloeber? There are millions of such individual computers.
Can't you release a new version of Eclipse or Sloeber with the Arduino or Sloeber plugin installed, like it was in Sloeber 4.1? That would solve a lot of problems.

**Comment by ianr34** [comment]
"Since 2022 most versions have problems with the Eclipse Arduino plugin"

Not true. I swapped to a new computer about 6 months ago (did a clean install of Windows 10), so installed it from scratch - 4.4.3 - and it works fine on any of the ardunio boards, and 8266 boards, and esp32 boards I use. Did you follow the install instructions?

**Comment by darek-i-rafal-i-zbychu** [comment]
Have you checked under Windows 11? Please provide links to files that work correctly. Did the Arduino plugin install without a problem and which version. Try the MegatinyCore package, does it work correctly?

**Comment by darek-i-rafal-i-zbychu** [comment]
The program, when compiling the project, generates an incorrect compilation files of the project: Relase\makefile and  Release\subdir.mk, which should contain the correct path to the compiler.  Unfortunately, this is not the case. the file contains the entry: "/bin/avr-g++". In the Windows version of the program. See how this file looks like in Sloeber 4.1. In this version it compiles correctly, because this version includes Eclipse-arduino-plugin version 4.4.0. only this old version works correctly. How long will we be forced to use the old versions of Sloeber and Eclipse, because the new ones do not work?
I found a solution to Eclipse issues for Arduino projects in Windows. Please see the website:
https://progel.pl/?mp=elektronika/programy/problemy_eclipse
